src/opentimelineio/trackAlgorithm.cpp

```diff
@@ -27,7 +27,12 @@ Track* track_trimmed_to_range(Track* in_track, TimeRange trim_range, ErrorStatus
         }
         
         auto child_range = child_range_it->second;
-        if (!trim_range.overlaps(child_range)) {
+      if ((trim_range.duration().value() == 0) ||
+            (!trim_range.overlaps(child_range) && !child_range.overlaps(trim_range) &&
+            !trim_range.begins(child_range) && !child_range.begins(trim_range) &&
+            !trim_range.finishes(child_range) && !child_range.finishes(trim_range) &&
+            !trim_range.contains(child_range) && !child_range.contains(trim_range) &&
+            !(trim_range == child_range))) {
```

Comment:
- 2 space -> 4 space indentation
- this condition suggests that in addition to the foundational predicates provided by the algebra, we should provide a few high level convenience predicates.  In this case, this block is asking the _very_ useful and common question "do these two time ranges share any time points".  I think that including an inclusive `intersects` function that implements this test would be beneficial.   The intersect test should be implemented as simply and directly as possible rather than using the existing predicates exclusively.